test/e2e/test/diagnostics.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/command"
+)
+
+var once sync.Once
+
+// canRunDiagnostics will determine if this e2e test run has the ability to run eck-diagnostics after
+// each test failure, which includes uploading the resulting zip file to a GS bucket. If the job name
+// is set (not empty), the google credentials file exists, and the eck-diagnostics binary exists
+// then we should be able to run diagnostics.
+func canRunDiagnostics(ctx Context) bool {
+	// If we don't have credentials to write to bucket, then we can't run diagnostics
+	// after e2e test failures, as we upload the zip file to a bucket.
+	if _, err := os.Stat(ctx.GCPCredentialsPath); err != nil && errors.Is(err, fs.ErrNotExist) {
+		return false
+	} else if err != nil {
+		log.Error(err, "while checking for existence of %s", ctx.GCPCredentialsPath)
+		return false
+	}
+	// If we're not in CI, then don't run diagnostics on e2e test failures.
+	if os.Getenv("CI") == "" {
```

Comment:
Like https://github.com/elastic/cloud-on-k8s/pull/6363/files#r1151097338.
```suggestion
	if os.Getenv("CI") != "true" {
```